crates/module-system/hyperlane/src/warp/hooks.rs

```diff
@@ -15,11 +15,21 @@ impl<S: Spec> Warp<S> {
         let route_ids = self.get_monitored_route_ids();
         let visible_slot = state.current_visible_slot_number();
 
+        tracing::trace!(
+            route_count = route_ids.len(),
+            "Emitting rate limiter metrics at slot {}",
```

Comment:
String interpolation will blow up structured loggin, such as loki

Reply:
should I use `format!` instead?

Reply:
> It's a high cardinality field, is it OK to add it as an attribute?

Yep, that's the idea.